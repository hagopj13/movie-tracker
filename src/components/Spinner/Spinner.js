import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: theme.spacing(3, 0),
  },
}));

const Spinner = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color="primary" />
    </div>
  );
};

export default Spinner;
