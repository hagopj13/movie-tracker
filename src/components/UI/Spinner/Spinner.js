import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: theme.spacing(3, 0),
  },
}));

export default function CircularIndeterminate() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color="primary" />
    </div>
  );
}
