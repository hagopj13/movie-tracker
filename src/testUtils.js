import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { createMount } from '@material-ui/core/test-utils';

import { initialState } from 'store/rootReducer';

const mount = createMount();
const mockStore = configureMockStore()(initialState);

export const mountWithStore = (children, store = mockStore) =>
  mount(<Provider store={store}>{children}</Provider>);
