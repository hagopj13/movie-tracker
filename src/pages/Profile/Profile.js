// @flow
import React, { useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import Spinner from 'components/Spinner/Spinner';
import ProfileActionTypes from 'store/user/profile/profile.types';
import profileActions from 'store/user/profile/profile.actions';
import loadingSelectors from 'store/api/loading/loading.selectors';
import loadedSelectors from 'store/api/loaded/loaded.selectors';
import ConfigActionTypes from 'store/config/config.types';

import ProfileHeader from './Header/ProfileHeader';
import ProfileTabs from './Tabs/ProfileTabs';

type Props = {
  isLoading: boolean,
  isLoaded: boolean,
  onFetchProfile: () => void,
};

export const Profile = (props: Props) => {
  const { isLoading, isLoaded, onFetchProfile } = props;

  useLayoutEffect(() => {
    onFetchProfile();
  }, [onFetchProfile]);

  if (isLoading || !isLoaded) {
    return <Spinner />;
  }

  return (
    <>
      <ProfileHeader />
      <ProfileTabs />
    </>
  );
};

const mapStateToProps = createStructuredSelector({
  isLoading: loadingSelectors.createIsLoadingSelector([
    ProfileActionTypes.FETCH_PROFILE,
    ConfigActionTypes.FETCH_IMAGES_CONFIG,
  ]),
  isLoaded: loadedSelectors.createIsLoadedSelector([ProfileActionTypes.FETCH_PROFILE]),
});

const mapDispatchToProps = {
  onFetchProfile: profileActions.fetchProfile,
};

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
