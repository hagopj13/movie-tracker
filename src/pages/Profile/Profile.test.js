import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { mountWithStore } from 'testUtils';
import Spinner from 'components/Spinner/Spinner';

import { Profile } from './Profile';

describe('Profile page', () => {
  let shallow;
  let wrapper;

  const mockFetchProfile = jest.fn();

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the Profile page correctly', () => {
    const mockProps = {
      isLoading: false,
      isLoaded: true,
      onFetchProfile: mockFetchProfile,
    };
    wrapper = shallow(<Profile {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Spinner if isLoading is true', () => {
    const mockProps = {
      isLoading: true,
      isLoaded: true,
      onFetchProfile: mockFetchProfile,
    };
    wrapper = shallow(<Profile {...mockProps} />);
    expect(wrapper.exists(Spinner)).toBe(true);
  });

  it('should render a Spinner if isLoaded is false', () => {
    const mockProps = {
      isLoading: false,
      isLoaded: false,
      onFetchProfile: mockFetchProfile,
    };
    wrapper = shallow(<Profile {...mockProps} />);
    expect(wrapper.exists(Spinner)).toBe(true);
  });

  it('should fetch profile after mounting', () => {
    const mockProps = {
      isLoading: false,
      isLoaded: true,
      onFetchProfile: mockFetchProfile,
    };
    wrapper = mountWithStore(<Profile {...mockProps} />);
    expect(mockFetchProfile).toHaveBeenCalledTimes(1);
  });
});
