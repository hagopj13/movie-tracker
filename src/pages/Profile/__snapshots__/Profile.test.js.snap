// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page should render the Profile page correctly 1`] = `
<Fragment>
  <Connect(ProfileHeader) />
  <ProfileTabs />
</Fragment>
`;
